app_feup/lib/controller/networking/network_router.dart

```diff
@@ -225,6 +226,78 @@ class NetworkRouter {
     return tripList;
   }
 
+  // Gets the next arrivals for a given bus stop
+  // Uses FlutterWebviewPlugin in order to scrap the information after the
+  // Javascript is loaded
+  static Future<List<Trip>> getNexArrivalsAfterLoadingJS(
+      String stopCode, BusStopData stopData) async {
+    final flutterWebviewPlugin = FlutterWebviewPlugin();
+
+    final Completer<List<Trip>> tripsCompleter = Completer();
+    final tripList = <Trip>[];
+
+    flutterWebviewPlugin.launch(
+        'https://www.stcp.pt/en/travel/timetables/?paragem=' +
+            stopCode +
+            '&t=smsbus',
+        hidden: true);
+
+    // When it finishes loading the page, should complete future for tripList
+    flutterWebviewPlugin.onStateChanged.listen((viewState) async {
+      if (viewState.type == WebViewState.finishLoad) {
+        flutterWebviewPlugin
+            .evalJavascript(
+                "document.querySelector('#paragem_info_result').innerHTML")
+            .then((html) {
+          final htmlNew = html
+              .replaceAll('\\t', '')
+              .replaceAll('\\n', '')
+              .replaceAll('\\u', '')
+              .replaceAll('\\', '')
+              .replaceAll('003C', '<')
+              .replaceAll('-', '')
+              .trim();
+
+          final htmlResponse = parseFragment(htmlNew);
+
+          final tableEntries =
+              htmlResponse.querySelectorAll('#smsBusResults > tbody > tr.even');
+
+          final configuredBuses = stopData.configuredBuses;
+
+          for (var entry in tableEntries) {
+            final rawBusInformation = entry.querySelectorAll('td');
+
+            final busLine =
+                rawBusInformation[0].querySelector('ul > li').text.trim();
+
+            if (!configuredBuses.contains(busLine)) {
+              continue;
+            }
+
+            final busDestination = rawBusInformation[0].text;
+
+            final busTimeRemaining = getBusTimeRemaining(rawBusInformation);
+            final Trip newTrip = Trip(
+                line: busLine,
+                destination: busDestination,
+                timeRemaining: busTimeRemaining);
+
+            tripList.add(newTrip);
+          }
```

Comment:
This can be done with a map followed by a `where` followed by another map, which would fit more into our functional style of writing code. Can you make that change?